models/mask_generator.py

```diff
@@ -298,13 +298,26 @@ def optimize_parameters(self, curr_iter=0):
         self.optimizer_D_P.zero_grad()
         self.backward_D_P(curr_iter)
         self.optimizer_D_P_step()
+    
+    def set_input_display(self, input):
+        #for image log
+        # Sim data
+        self.image = input.data.x.unsqueeze(0).to(self.device)
+        mask = input.data.m.to(self.device)
+        self.mask = mask.unsqueeze(0)
```

Comment:
I'm just bugging you but coding wise it'd be better to deal with the same procedure identically, meaning 

```
self.mask = input.data.m.unsqueeze(0).to(self.device)
```

Reply:
Done thanks!